core/arch/x86_64/mm/core_mmu.c

```diff
@@ -865,7 +865,12 @@ static void assign_mem_granularity(struct tee_mmap_region *memory_map)
 
 static unsigned int get_va_width(void)
 {
-	return 64;
+	uint32_t reg_a;
+	uint32_t unused;
+	cpuid(X86_CPUID_ADDR_WIDTH, &reg_a, &unused, &unused, &unused);
```

Comment:
There is already x86_get_address_width function or you can return g_vaddr_width directly.